### PR TITLE
[DONE] Add scenario download feature to result selection tool.

### PR DESCRIPTION
**Pull request by jackieleng**
See changes.

### CODE REVIEW

**Comment by jackieleng** [comment]
[[QGIS plugin] - design, estimate and make user friendly download raw results functionality in plugin [5]](https://trello.com/c/SGC3eeQ9/1776-qgis-plugin-design-estimate-and-make-user-friendly-download-raw-results-functionality-in-plugin-5)

File: threedi_result_selection.py [hunk 1]
```diff
@@ -115,3 +142,76 @@ def get_state_description(self):
...
+                self.on_single_download_ready_to_read_chunk)
+            reply.finished.connect(self.on_single_download_finished)
+        pop_up_info("Download started.")
+
+    def on_single_download_ready_to_read_chunk(self):
+        """Process a chunk of the downloaded data."""
+        reply = self.sender()
+        raw_chunk = reply.readAll()  # QByteArray
+        filename = reply.url().toString().split('/')[-1]
+        with open(os.path.join(self.download_directory, filename), 'ab') as f:
```
Comment (jackieleng) [inline threedi_result_selection.py]: Note to self: `self.download_directory` will change when a new scenario download is started. `QNetworkRequest.setAttribute` is perhaps an idea.

File: threedi_result_selection.py [hunk 2]
```diff
@@ -29,6 +44,16 @@ def __init__(self, iface, ts_datasource):
         self.iface = iface
 
         self.ts_datasource = ts_datasource
+        # TODO: unsure if this is the right place for initializing this model
+        self.download_result_model = DownloadResultModel()
```
Comment (larsclaussen) [inline threedi_result_selection.py:48]: kan de TODO weg?

**Review by larsclaussen — COMMENTED** [review]
ziet er verder wel goed uit hoor

**Pull request merged**